appinventor/blocklyeditor/src/drawer.js

```diff
@@ -81,6 +87,209 @@ Blockly.Drawer.buildTree_ = function() {
   return tree;
 };
 
+Blockly.Drawer.createSubsetBlockInfoArray_ = function() {
+  console.log("HELLO creating subset block info array here");
+  try {
+    console.log("trying to set subset string");
+    var subsetJsonString;
+    var formName = Blockly.mainWorkspace.formName;
+    var screenName = formName.split("_")[1];
+    if (window.parent.BlocklyPanel_getComponentInstancePropertyValue) {
+      subsetJsonString = window.parent.BlocklyPanel_getComponentInstancePropertyValue(formName, screenName, "BlocksToolkit");
+    } else {
+      subsetJsonString = JSON.stringify(window.reactGetSubsetString());
+    }  
+    console.log("subsetJsonString");
+    console.log(subsetJsonString);
+    var subsetArray = JSON.parse(subsetJsonString);
```

Comment:
If subsetJsonString is the empty string (which is the case when loading an older project) then this line throws and exception. Should check to see if the string is empty and provide an appropriate define ([] or {}?).

Reply:
Heh, and I just hit this when testing. Working on it.